pkg/npm/http-api/src/Urbit.ts

```diff
@@ -172,17 +189,22 @@ export class Urbit {
       method: 'post',
       body: `password=${this.code}`,
       credentials: 'include',
-    }).then((response) => {
+    }).then(async response => {
       if (this.verbose) {
         console.log('Received authentication response', response);
       }
-      const cookie = response.headers.get('set-cookie');
-      if (!this.ship) {
-        this.ship = new RegExp(/urbauth-~([\w-]+)/).exec(cookie)[1];
+
+      if (response.status !== 204) {
```

Comment:
Feels like we should just check for `>= 200` and `< 300` here...

Reply:
Fixed.